lib/narwin_chat/talks.ex

```diff
@@ -0,0 +1,103 @@
+defmodule NarwinChat.Talks do
+  require Logger
+
+  import Ecto.Query
+
+  alias NarwinChat.Repo
+  alias NarwinChat.Chat.Room
+  alias NarwinChat.Chat.Talk
+
+  def list_current_talks(opts \\ []) do
+    now = opts[:at] || DateTime.utc_now()
+
+    Talk
+    |> where([t], t.starts_at <= ^now and t.ends_at > ^now)
+    |> Repo.all()
+    |> Repo.preload([:room])
```

Comment:
```suggestion
    |> where([t], t.starts_at <= ^now and t.ends_at > ^now)
    |> preload([t], [:room])
    |> Repo.all()
```

I'd do it all in one DB query, since this is running very frequently

Reply:
I tested this locally and it still runs two DB queries. If there was a way to make it do one DB query via joins, that would be cool to know, but I don't think preload works that way.

Reply:
Oh, I just assumed it joined by default. I think you could do it like this (untested), but I leave it to you whether it's important.

```suggestion
    |> where([t], t.starts_at <= ^now and t.ends_at > ^now)
    |> join(:inner, [t], r in Room, on: t.room_id == r.id)
    |> preload([t, r], [room: r])
    |> Repo.all()
```